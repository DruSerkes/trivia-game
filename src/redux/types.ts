export const ADD_QUESTIONS = 'ADD_QUESTIONS';
export const REMOVE_QUESTIONS = 'REMOVE_QUESTIONS';
export const QUESTIONS_FAIL = 'QUESTIONS_FAIL';

interface Question {
	question: string;
	correct_answer: string;
	incorrect_answers: string[];
	[propName: string]: any;
}

interface AddQuestions {
	type: typeof ADD_QUESTIONS;
	payload: Question[];
}

interface QuestionsFail {
	type: typeof QUESTIONS_FAIL;
}

interface RemoveQuestions {
	type: typeof REMOVE_QUESTIONS;
}

export type QuestionType = Question;
export type QuestionsDispatchTypes = AddQuestions | QuestionsFail | RemoveQuestions;
