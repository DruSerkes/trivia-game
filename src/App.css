.App {
	text-align: center;
	background-color: #181818;
	color: whitesmoke;
	height: 100vh;
}

.Start {
	position: relative;
	top: 100px;
	width: 30%;
	margin: 0 auto;
}

.Start button {
	font-size: 1.5rem;
}

.QuizCard {
	background-color: dimgray;
    width: 69%;
    margin: 0 auto;
    padding: 1rem 1rem 0;
    border: 2px solid white;
    border-radius: 5px;
}

.Answer {
	cursor: pointer;
}

.AnswerText {
	margin: 1.25rem auto;
    font-size: 19px;
    color: lawngreen;
}

.Result {
	margin: 2rem auto 0;
    min-height: 200px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
}

@keyframes App-logo-spin {
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
}


@media screen and (max-width: 380px){
	.Start {
		width: 70%;
	}
	.Start button {
		font-size: 1rem;
	}
}