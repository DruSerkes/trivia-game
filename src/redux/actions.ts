import { ADD_QUESTIONS, QuestionType, QuestionsDispatchTypes, QUESTIONS_FAIL, REMOVE_QUESTIONS } from './types';
import { Dispatch } from 'redux';

const BASE_URL = `https://opentdb.com/api.php`;

export function getQuestions(amount = '10', difficulty?: string, type = 'multiple',) {
	return async function (dispatch: Dispatch) {
		try {
			if (difficulty) {
				const response = await fetch(`${BASE_URL}?amount=${amount}&type=${type}&difficulty=${difficulty}`);
				const data = await response.json();
				dispatch(gotQuestions(data));
			} else {
				const response = await fetch(`${BASE_URL}?amount=${amount}&type=${type}`);
				const data = await response.json();
				dispatch(gotQuestions(data));
			}

		} catch (e) {
			console.log(e);
			dispatch(questionsFail());
		}
	};
}

export function gotQuestions(questionData: any): QuestionsDispatchTypes {
	const payload = questionData.results.map((data: QuestionType) => {
		const { question, correct_answer, incorrect_answers }: QuestionType = data;
		return { question, correct_answer, incorrect_answers };
	});
	return {
		type: ADD_QUESTIONS,
		payload
	};
}

export function removeQuestions(): QuestionsDispatchTypes {
	return {
		type: REMOVE_QUESTIONS
	}
}

export function questionsFail(): QuestionsDispatchTypes {
	return {
		type: QUESTIONS_FAIL
	};
}
