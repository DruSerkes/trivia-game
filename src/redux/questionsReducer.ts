import { ADD_QUESTIONS, QuestionType } from './types';

const INITIAL_STATE: QuestionType[] = [];

const questionsReducer = (state = INITIAL_STATE, action: any) => {
	switch (action.type) {
		case ADD_QUESTIONS:
			return [ ...action.payload ];
		default:
			return state;
	}
};

export default questionsReducer;
