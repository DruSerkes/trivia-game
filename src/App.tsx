import React from 'react';
import './App.css';
import { TriviaGame } from './components/TriviaGame';
import { Typography, Box } from '@material-ui/core'

/* TODO 
1a. Debug background color staying the same when you click for True/False questions 
2. Add feedback text for correct / incorrect 
3. Refactor to show which was correct if incorrect was chosen... (give it a 1px green border)
*/
function App() {
      return (
            <Box className="App">
                  <Typography gutterBottom variant="h1">Trivia Game!</Typography>
                  <TriviaGame />
            </Box>);
}

export default App;
