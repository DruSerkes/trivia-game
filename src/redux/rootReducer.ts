import { questionsReducer } from './questionsReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ questions: questionsReducer });

export default rootReducer;
