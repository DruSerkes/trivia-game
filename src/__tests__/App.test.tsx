import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../redux/rootReducer'
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));


test('App renders without breaking ', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<App />
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
