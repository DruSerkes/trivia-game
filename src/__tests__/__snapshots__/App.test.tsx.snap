// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without breaking  1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1 App"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom"
    >
      Trivia Game!
    </h1>
    <div
      class="MuiBox-root MuiBox-root-2 TriviaGame"
    >
      <div
        class="MuiBox-root MuiBox-root-3 Start"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Start New Game
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
